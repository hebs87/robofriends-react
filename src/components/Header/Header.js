import {Component} from 'react';

class Header extends Component {
  // Determine when to render the header - only render once as it doesn't need to re-render
  shouldComponentUpdate(nextProps, nextState, nextContext) {
    return false;
  }

  render() {
    return (
      <>
        <h1 className='f1'>RoboFriends</h1>
      </>
    )
  }
}

export default Header;
