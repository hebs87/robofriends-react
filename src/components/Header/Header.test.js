import {shallow} from 'enzyme';
import Header from "./Header";

describe('Header component test suite', () => {
  it('expect to render Header component', () => {
    // shallow renders just the target component and not any nest components that are rendered within it
    // toMatchSnapshot() takes a snapshot of the component and tests it is rendering as it should - any change to the
    // component will then cause the test to fail, as it will no longer match the snapshot
    expect(shallow(<Header/>)).toMatchSnapshot();
  });
});
