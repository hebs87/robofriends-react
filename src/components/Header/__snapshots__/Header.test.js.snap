// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component test suite expect to render Header component 1`] = `
<Fragment>
  <h1
    className="f1"
  >
    RoboFriends
  </h1>
</Fragment>
`;
